### PR TITLE
create a Dockerfile, and make Travis push images

### PR DESCRIPTION
**Pull request by oconnor663**
creating PR to trigger Travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,18 +11,31 @@ addons:
...
-  - (cd protocol && npm i && git diff --quiet --exit-code)
 matrix:
   - os: osx
 branches:
   only:
     - master
+services:
+  - docker
+before_install:
+  - mysql -e "create database IF NOT EXISTS test;" -uroot
```
Comment (jzila) [inline .travis.yml]: is this a local mysql install?

File: .travis.yml [hunk 2]
```diff
@@ -11,18 +11,31 @@ addons:
...
+before_script:
+  - (cd protocol && npm i && git diff --quiet --exit-code)
```
Comment (jzila) [inline .travis.yml]: what does this do?

File: .travis.yml (hunk 1 above)
Comment (oconnor663) [inline .travis.yml]: I think so. I just moved this line from above, to make the order of operations look clearer to me.

File: .travis.yml (hunk 2 above)
Comment (oconnor663) [inline .travis.yml]: This is checking to make sure that the generated-and-checked-in protocol files haven't gotten out of date compared to the AVDL files they're generated from. This is another section that I just moved down from above.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu:14.04
+MAINTAINER Keybase <admin@keybase.io>
+
+RUN useradd -m keybase
+USER keybase
+ENV PATH=/home/keybase:$PATH
+WORKDIR /home/keybase
+
+COPY ./gregord/gregord /home/keybase/gregord
```
Comment (jzila) [inline Dockerfile]: This assumes `go build` and not `go install`, correct?

**Comment by jzila** [comment]
👍

File: Dockerfile (hunk 3 above)
Comment (oconnor663) [inline Dockerfile]: Yep, that's the `(cd gregord && go build)` above.

**Pull request merged**